File: bookstore-server/src/main/java/com/nixsolutions/bondarenko/bookstore/entity/Book.java
package com.nixsolutions.bondarenko.bookstore.entity;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Book implements Serializable {
  private static final long serialVersionUID = 5404904932186585449L;

  @Id
  @GeneratedValue(strategy = GenerationType.SEQUENCE)
  private long id;

  @Column(nullable = false)
  private String name;

  @Column(nullable = true)
  private String photo;

  @Column(columnDefinition = "boolean default false", nullable = false)
  private boolean absent;

  @Column(nullable = false)
  private Double price;
}

File: bookstore-client/src/app/entity/book.ts
import { Entity } from './entity';

export class Book  implements Entity{
  collectionName = 'books';

  constructor(public id: number = null,
              public name: string = null,
              public price: number = null,
              public photo: string = null,
              public absent: boolean = false,
              public _links: any = null) {
  }
}
